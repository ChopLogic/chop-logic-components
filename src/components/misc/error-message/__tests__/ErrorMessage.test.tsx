import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import ChopLogicErrorMessage from 'components/misc/error-message/ErrorMessage';

describe('ChopLogicErrorMessage', () => {
  const testProps = {
    errorId: 'test-error-id',
    visible: true,
    message: 'Incorrect input',
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(<ChopLogicErrorMessage {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain the test id', () => {
    const testId = 'test-message-id';
    render(<ChopLogicErrorMessage {...testProps} testId={testId} />);
    expect(screen.getByTestId(testId)).toBeInTheDocument();
  });
});
