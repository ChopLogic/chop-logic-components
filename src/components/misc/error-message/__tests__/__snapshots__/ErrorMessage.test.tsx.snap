// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicErrorMessage > should match the snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-braxZu jjfUlA"
    id="test-error-id"
  >
    Incorrect input
  </span>
</DocumentFragment>
`;
